steely/plugins/emoji.py

```diff
@@ -28,37 +27,54 @@
     emoji = EMOJI_DATA[k]['char']
     if emoji is None:
         continue
-    WORD_TO_EMOJI[k].append(emoji)
+
+    # Without this if check, 'it' stil maps to an italian flag.
+    if not (len(k) == 2 and EMOJI_DATA[k]['category'] == 'flags'):
+        WORD_TO_EMOJI[k].append(emoji)
+    
     for word in EMOJI_DATA[k]['keywords']:
         if word in BLACKLIST:
             continue
+        if len(word) == 2 and EMOJI_DATA[k]['category'] == 'flags':
+            continue
         WORD_TO_EMOJI[word].append(emoji)
 
 
 def emojify_or_not_i_am_a_function_not_a_cop(word):
-    word = word.lower()
-    if word in WORD_TO_EMOJI:
-        e = random.choice(WORD_TO_EMOJI[word]) * \
-            random.choice([1, 1, 1, 2, 2, 3])
+    w = word.lower()
+    w = ''.join(c for c in w if c.isalnum())
+    if not len(w):
+        return word
+    if w not in WORD_TO_EMOJI and w[-2:] == 'es':
+        w = w[:-2] # peaches -> peach
+    if w not in WORD_TO_EMOJI and w[-1] == 's':
+        w = w[:-1] # aubergines -> aubergine
+    if w not in WORD_TO_EMOJI and w[-3:] == 'ing':
+        w = w[:-3] # hating -> hat
```

Comment:
I would probably smack these in a list and use `w.startswith(thing)`

Reply:
.mock

Reply:
I think putting these in a list and doing w.endswith would be less clear. They all handle specific cases (and there were other cases I didn't bother adding next to them. Like I was debating "running" -> "runn" -> "run", but cba) that are more clearly laid out like this. Plus, the order they are run is actually important (ie. removing "s" from "peaches" would be useless), and putting them in a list wouldn't make it as clear not to mix them up later.

Ideally, I want to do some real NLP to map plurals to singulars (so "men" becomes "man" too), but that will come later.